### PR TITLE
Being buckled to a bed now saves you from "the floor is lava"

### LINKED ISSUE #33194: "The floor is lava" event does not treat beds as surfaces you can be on top of.
Issue: Being on a bed during a floor is lava event does not protect you from lava
How to repeat: buckle yourself to a bed and cause the lava floor event.  Die to lava

It would make sense that chairs/beds/etc should probably all protect from the floor is lava event, provided the user is strapped onto them.

Issue reported from Round ID: 79491 (Sybil)

[Directions]: # (Include the Round ID from the Status panel or retrieve it from https://atlantaned.space/newSS13tools/round.php ! If you believe the issue to be caused by a testmerge [OOC tab -> Show Server Revision], report it in the pull request's comment section instead. Explain your issue in detail, including the steps to reproduce it.)

[For Admins]: # (Oddities induced by var-edits and other admin tools are not necessarily bugs. Verify that your issues occur under regular circumstances before reporting them.)

image

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl:
fix: You can now lay (buckle!) yourself to a bed to avoid being burnt to a crisp during "the floor is lava" event.
/:cl:

fixes #33194

@Spaceinaba also fixes resin

### CODE REVIEW

**Comment by Spaceinaba** [comment]
🙏 god bless

**Comment by praisenarsie** [comment]
what about chairs? what about etc?

**Comment by ExcessiveUseOfCobblestone** [comment]
eh okay

**Comment by Supermichael777** [comment]
To burn maint workers to a crisp duh

**Comment by epochayur** [comment]
does this include rollerbeds and the like? imo anything you can buckle to should work

**Comment by ExcessiveUseOfCobblestone** [comment]
Any bed structure.

> does this include rollerbeds and the like? imo anything you can buckle to should work

I was going to do that but I was too lazy to add an exception to pipes lol

File: code/datums/weather/weather_types/floor_is_lava.dm [hunk 1]
```diff
@@ -20,10 +20,11 @@
 
 	overlay_layer = ABOVE_OPEN_TURF_LAYER //Covers floors only
 	immunity_type = "lava"
+	
 
 /datum/weather/floor_is_lava/weather_act(mob/living/L)
-	for(var/obj/structure/O in L.loc)
-		if(O.density)
+	for(var/obj/structure/O in L.loc)			
+		if(O.density || (istype(O, /obj/structure/bed) && (L in O.buckled_mobs)))
```
Comment (Cyberboss) [inline code/datums/weather/weather_types/floor_is_lava.dm]: reverse the order of the buckled/bed checks

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
Travis

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**